Doc/library/typing.rst

```diff
@@ -2429,6 +2429,72 @@ Functions and decorators
 
    .. versionadded:: 3.11
 
+.. decorator:: dataclass_transform
+
+   The :data:`~typing.dataclass_transform` annotation may be used to
+   decorate a function that is itself a decorator, a class, or a metaclass.
+   The presence of ``@dataclass_transform()`` tells a static type checker that the
+   decorated function, class, or metaclass performs runtime "magic" that
+   transforms a class, endowing it with dataclass-like behaviors.
+
+   Example usage with a decorator function::
+
+      _T = TypeVar("_T")
+
+      @dataclass_transform()
+      def create_model(cls: type[_T]) -> type[_T]:
```

Comment:
I think it looks less alien to new users of `typing` (who this feature probably won't be targeted towards, but still) if we don't use leading underscores for `TypeVar` names

```suggestion
      T = TypeVar("T")

      @dataclass_transform()
      def create_model(cls: type[T]) -> type[T]:
```